### PR TITLE
feat(marxan-run): allow to cancel running process

### PR DESCRIPTION
**Pull request by kgajowy**
This PR revisits Marxan run:
- make PID not being necessary and keep the "process" reference
- introduce "cancellables" which react on kill signal
- uses AbortController (polyfill, as natively its available since node v16)
- uses `ModuleRef` to create instances of underlying classes per-run basis. 

TODO:
- [x] clear AbortController on finish/kill/error

### CODE REVIEW

**Comment by kgajowy** [comment]
@hotzevzl still needs `clear AbortController on finish/kill/error` to be done but mainly can be reviewed.

File: api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts [hunk 1]
```diff
@@ -8,27 +8,66 @@ import { v4 } from 'uuid';
...
+beforeEach(async () => {
   fixtures = await getFixtures();
 });
 
+describe(`given input data is delayed`, () => {
+  beforeEach(() => {
+    fixtures.GivenInputFilesAreAvailable(5000);
+  });
+
+  test(`cancellable slow marxan run`, async (done) => {
```
Comment (hotzevzl) [inline api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts:24]: can we improve the description of this test?

File: api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts [hunk 2]
```diff
@@ -8,27 +8,66 @@ import { v4 } from 'uuid';
...
     fixtures.GivenInputFilesAreAvailable();
   });
-
   test(`marxan run`, async () => {
     await fixtures.WhenRunningMarxan();
 
     expect(await fixtures.ThenExecutionOutput()).toBeGreaterThan(0);
   }, 30000);
+
+  test(`cancellable marxan run`, async (done) => {
```
Comment (hotzevzl) [inline api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts:52]: same as above

**Review by hotzevzl — APPROVED** [review]

File: api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts [hunk 3]
```diff
@@ -8,27 +8,66 @@ import { v4 } from 'uuid';
...
+describe(`given input data is delayed`, () => {
+  beforeEach(() => {
+    fixtures.GivenInputFilesAreAvailable(5000);
+  });
+
+  test(`cancelling marxan run during fetching assets`, async (done) => {
+    expect.assertions(1);
+
+    fixtures
+      .WhenRunningMarxan()
```
Comment (Dyostiq) [inline api/apps/geoprocessing/test/integration/marxan-run/marxan-run.e2e-spec.ts:28]: actually it's `GivenRunningMarxan`

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/marxan-sandbox-runner.service.ts [hunk 4]
```diff
@@ -1,57 +1,105 @@
...
+        this.markExecutionAsEnded(forScenarioId);
+        reject(error);
+      }
+    });
+  }
 
-      this.marxanRunner.execute(marxanBinaryPath, workingDirectory);
+  private markExecutionAsEnded(ofScenarioId: string) {
+    delete this.#controllers[ofScenarioId];
+  }
```
Comment (Dyostiq) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/marxan-sandbox-runner.service.ts:89]: `clearAbortController`?

**Review by Dyostiq — CHANGES_REQUESTED** [review]

**Review by Dyostiq — APPROVED** [review]

**Pull request merged**